src/fileio/datafile.cxx

```diff
@@ -88,12 +88,14 @@ Datafile& Datafile::operator=(const Datafile &rhs) {
   f3d_arr      = rhs.f3d_arr;
   v2d_arr      = rhs.v2d_arr;
   v3d_arr      = rhs.v3d_arr;
-  filenamelen=FILENAMELEN;
-  filename     = new char[filenamelen];
   return *this;
 }
 
 Datafile::~Datafile() {
+  if (file){
+    delete file;
+    file = NULL;
```

Comment:
I could, but the code uses NULL every where else in this file.
Should this be changed everywhere?
Probably the whole code should be checked for this. Usage of NULL: 1282. Usage of nullptr: 63
Isn't the only difference the one is already casted to pointer type?